lib/generators/test_track/migration_generator.rb

```diff
@@ -1,38 +1,77 @@
-require 'rails/generators/named_base'
+require 'rails/generators/base'
 
 module TestTrack
   module Generators
-    class MigrationGenerator < Rails::Generators::NamedBase
-      desc "Creates a test track migration file. Files that start with retire or finish will create migrations that finish a split."
+    class MigrationGenerator < Rails::Generators::Base
+      desc "Creates a test track migration file."
+
+      argument :raw_split_name, required: true
 
       def create_test_track_migration_file
-        create_file "db/migrate/#{formatted_time_stamp}_#{file_name}.rb", <<-FILE.strip_heredoc
-          class #{file_name.camelize} < ActiveRecord::Migration
+        create_migration_file
+      end
+
+      private
+
+      def create_migration_file
+        create_file full_file_path, <<-FILE.strip_heredoc
+          class #{split_class_name} < ActiveRecord::Migration
             def change
               TestTrack.update_config do |c|
-                #{split_command} :#{split_name}
+                #{split_command_line}
               end
             end
           end
         FILE
       end
 
-      private
+      def split_class_name
+        split_name.camelize
+      end
+
+      def full_file_path
+        "db/migrate/#{formatted_time_stamp}_#{split_name}.rb"
+      end
+
+      def split_command_line
+        "#{split_command} :#{split_name.gsub('finish_', '')}#{split_variants}"
```

Comment:
`split_name` doesn't seem like the right name for this method if it potentially has `finish_` on the front, right? Maybe more methods that do more different slicing and dicing?